src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -2507,18 +2501,19 @@ public Long zcount(final byte[] key, final byte[] min, final byte[] max) {
       final int offset, final int count) {
     checkIsInMultiOrPipeline();
     client.zrangeByScoreWithScores(key, min, max, offset, count);
-    return getBinaryTupledSet();
+    return getTupledSet();
   }
 
-  private Set<Tuple> getBinaryTupledSet() {
+  protected Set<Tuple> getTupledSet() {
+    checkIsInMultiOrPipeline();
```

Comment:
Do we need to check this again while reading response from Redis?

Reply:
No, we don't. Nice finding @HeartSaVioR . It's happened because PR is crafted before merging #1600 .